### PR TITLE
Ecs ec2 cluster

### PR DESCRIPTION
**Pull request by bmwosz**
works on mx-tools and storybook without issues
also I have created ecs cluster on testing account with that module.

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -44,11 +44,126 @@ module "rds" {
...
+  statement {
+    actions         = ["sts:AssumeRole"]
+    principals {
+      type = "Service"
+      identifiers = ["ec2.amazonaws.com"]
+    }
+  }
+}
+
+resource "aws_iam_role" "ecs_ec2_role" {
```
Comment (Maciej-Gierwatowski) [inline main.tf:74]: Add it's name/arn to outputs, can be usefull

File: main.tf [hunk 2]
```diff
@@ -44,11 +44,126 @@ module "rds" {
...
+  count               = var.launch_type == "EC2" ? 1 : 0
+  name                = "${module.label.id}-ec2-role"
+  assume_role_policy  = data.aws_iam_policy_document.ec2_role_document.json
+  tags                = module.label.tags
+}
+
+resource "aws_iam_instance_profile" "ecs_ec2_instance_profile" {
```
Comment (Maciej-Gierwatowski) [inline main.tf:81]: Add it's name/arn to outputs, can be usefull

File: main.tf [hunk 3]
```diff
@@ -44,11 +44,126 @@ module "rds" {
...
+  image_id                      = data.aws_ami.vm_ami.id
+  instance_type                 = var.instance_type
+  iam_instance_profile          = join("",aws_iam_instance_profile.ecs_ec2_instance_profile.*.arn)
+  user_data                     = templatefile(
+    "${path.module}/cloud-config.yml",
+    {
+      ecs_cluster_name = "${aws_ecs_cluster.default.name}"
+    }
+  )
+  associate_public_ip_address   = true
```
Comment (Maciej-Gierwatowski) [inline main.tf:140]: What if instance is launched in private subnets?

Comment (bmwosz) [reply main.tf:140]: Good point. I forgot to change that.

File: main.tf (hunk 1 above)
Comment (bmwosz) [reply main.tf:74]: I will print `name` and `arn` of the resource

File: main.tf (hunk 2 above)
Comment (bmwosz) [reply main.tf:81]: I will print `name` and `arn` of the resource

File: main.tf [hunk 4]
```diff
@@ -44,11 +44,136 @@ module "rds" {
...
+  role          = join("",aws_iam_role.ecs_ec2_role.*.name)
+  for_each      = var.launch_type == "EC2" ? toset(local.ecs_ec2_role_policies_list) : toset([])
+  policy_arn    = each.value
+}
+
+data "aws_ami" "vm_ami" {
+  most_recent = true
+  filter {
+    name   = "name"
+    values = ["amzn2-ami-ecs-hvm-2.0*"]
```
Comment (Maciej-Gierwatowski) [inline main.tf:97]: How about parametrizing this?

Comment (bmwosz) [reply main.tf:97]: no problem

Comment (bmwosz) [reply main.tf]: I pushed and check changes. It should be fine right now

**Pull request merged**